avocado/utils/configure_network.py

```diff
@@ -35,168 +37,248 @@ class NWException(Exception):
-            return True
-        except process.CmdError as ex:
-            raise NWException("ifup fails: %s" % ex)
+    def __init__(self, if_name, remote_session=None):  # pylint: disable=W0231
+        self.name = if_name
+        self.remote_session = remote_session
 
+    def set_ip(self, ipaddr, netmask, interface_type=None):
+        """
+        utility assign a IP  address (given to this utility ) to  Interface
+        and gernerate interface file based on distro
+        :param ipaddr : ip address which need to configure for interface
+        :param netmask: Netmask which is associated  to provided IP
+        :param interface_type: Interface type IPV4 or IPV6 , default is
+                               IPV4 style
+        """
 
-def unset_ip(interface):
-    """
-    Gets interface name unassigns the IP to the interface
-    """
-    if distro.detect().name == 'rhel':
-        conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s" % interface
+        if distro.detect().name == 'rhel':
+            conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s" % self.name
+            self._move_config_file(conf_file, "%s.backup" % conf_file)
+            with open(conf_file, "w") as network_conf:
+                if interface_type:
+                    interface_type = 'Ethernet'
+                network_conf.write("TYPE=%s \n" % interface_type)
+                network_conf.write("BOOTPROTO=none \n")
+                network_conf.write("NAME=%s \n" % self.name)
+                network_conf.write("DEVICE=%s \n" % self.name)
+                network_conf.write("ONBOOT=yes \n")
+                network_conf.write("IPADDR=%s \n" % ipaddr)
+                network_conf.write("NETMASK=%s \n" % netmask)
+                network_conf.write("IPV6INIT=yes \n")
+                network_conf.write("IPV6_AUTOCONF=yes \n")
+                network_conf.write("IPV6_DEFROUTE=yes")
+
+        if distro.detect().name == 'SuSE':
+            conf_file = "/etc/sysconfig/network/ifcfg-%s" % self.name
+            self._move_config_file(conf_file, "%s.backup" % conf_file)
+            with open(conf_file, "w") as network_conf:
+                network_conf.write("IPADDR=%s \n" % ipaddr)
+                network_conf.write("NETMASK='%s' \n" % netmask)
+                network_conf.write("IPV6INIT=yes \n")
+                network_conf.write("IPV6_AUTOCONF=yes \n")
+                network_conf.write("IPV6_DEFROUTE=yes")
+        self.bring_up()
 
-    if distro.detect().name == 'SuSE':
-        conf_file = "/etc/sysconfig/network/ifcfg-%s" % interface
+    def unset_ip(self):
+        """
+        Utility to unassign IP to Defined Interface
+        """
+        if distro.detect().name == 'rhel':
+            conf_file = "/etc/sysconfig/network-scripts/ifcfg-%s" % self.name
 
-    cmd = "ifdown %s" % interface
-    try:
-        process.system(cmd, sudo=True)
-        shutil.move(conf_file+".backup", conf_file)
+        if distro.detect().name == 'SuSE':
```

Comment:
You don't need to call `detect()` method every time inside the if. You could avoid multiple calls with:
```python
distro = distro.detect().name
if distro == 'SuSe':
   print('suse')
elif distro == 'rhel':
   print('rhel')
else:
   raise Exception("Distro Unknow, could not set ip")
```